sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/NetworkCallError.java

```diff
@@ -28,47 +33,131 @@ public NetworkCallError() {
+     */
+    public NetworkCallError className(Class<?> className) {
+        this.className = className;
+        return this;
+    }
 
-            case "java.lang.IndexOutOfBoundsException":
-                return new IndexOutOfBoundsException(throwable.getMessage());
+    /**
+     * Gets the name of the class of the throwable. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @return The class name which is Class type.
+     */
+    public Class<?> className() {
+        return this.className;
+    }
 
-            case "java.net.UnknownHostException":
-                return new UnknownHostException(throwable.getMessage());
+    /**
+     * Sets the throwable argument types which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @param argTypes The array of argument types.
+     * @return The instance of NetworkCallError class.
+     */
+    public NetworkCallError argTypes(Class<?>[] argTypes) {
+        this.argTypes = ImplUtils.clone(argTypes);
+        return this;
+    }
 
-            default:
-                return throwable;
-        }
+    /**
+     * Gets the throwable argument types which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
+     *
+     * @return The array of argument types.
+     */
+    public Class<?>[] argTypes() {
+        return ImplUtils.clone(this.argTypes);
     }
 
     /**
-     * Sets the throwable that was thrown during a network call.
+     * Sets the throwable argument default values which needed for new instance constructor. This is used during deserialization the construct the throwable
+     * as the actual class that was thrown.
      *
-     * @param throwable Throwable that was thrown.
+     * @param argValues The array of argument values.
+     * @return The instance of NetworkCallError class.
      */
-    public void throwable(Throwable throwable) {
-        this.throwable = throwable;
+    public NetworkCallError argValues(Object[] argValues) {
+        this.argValues = ImplUtils.clone(argValues);
+        return this;
     }
 
     /**
-     * Sets the name of the class of the throwable. This is used during deserialization the construct the throwable
+     * Gets the throwable argument default values which needed for new instance constructor. This is used during deserialization the construct the throwable
      * as the actual class that was thrown.
      *
-     * @param className Class name of the throwable.
+     * @return The array of argument values.
      */
-    public void className(String className) {
-        this.className = className;
+    public Object[] argValues() {
+        return ImplUtils.clone(this.argValues);
+    }
+
+    private Object getDefaultValue(Class<?> type) throws Exception {
```

Comment:
Do we want to let the exception bubble up as is or wrap it in a `RuntimeException`?

Reply:
The exception will be caught in NetworkCallError constructor and throw the error message which help to debug.